### PR TITLE
Use weighted matching algorithm from a package

### LINKED ISSUE #12: Use weighted graph matching algorithm
Weights for: 
* Last matches
* Same locations
* Same teams

Graph matching: https://www.boost.org/doc/libs/1_73_0/libs/graph/doc/maximum_weighted_matching.html

consider using Native Extensions for node to use Boost library for graphs https://medium.com/jspoint/a-simple-guide-to-load-c-c-code-into-node-js-javascript-applications-3fcccf54fd32

### PR DESCRIPTION
**Pull request by kelostrada**
Resolves #12

### CODE REVIEW

File: src/matching.js [hunk 1]
```diff
@@ -355,27 +25,36 @@ function buildGraph(matches, people) {
...
 
-      if (!matched) {
-        graph.push([i + 1, j + 1]);
-        edges++;
+      weight = !matched ? 10 : 0;
+      weight += people[i].location != people[j].location ? 5 : 0;
+      weight += people[i].team != people[j].team ? 5 : 0;
+      if (weight > 0) {
+        graph.push([i, j, weight]);
       }
```
Comment (agusmoles) [inline src/matching.js:33]: Why it should not be added to the graph if weight is 0? I think I could not satisfy all this `if` statements but still want to talk to the other person

File: test/matching.js [hunk 2]
```diff
@@ -1,8 +1,8 @@
 var assert = require("assert");
 const { matchPeople } = require("../src/matching");
 
-function person(id, added = true) {
-  return { id: id, added: added };
+function person(id, added = true, location = "", team = "") {
+  return { id: id, added: added, location: location, team: team };
 }
```
Comment (agusmoles) [inline test/matching.js:6]: In JS if variable is equal to the name of the property you can do shorthand assignment like this, and also maybe change to arrow func:

```suggestion
const person = (id, added = true, location = "", team = "") => (
  { id, added, location, team }
)
```

File: src/matching.js (hunk 1 above)
Comment (agusmoles) [reply src/matching.js:33]: Oh I missread, anyway you should consider the case where all people have talked. Maybe leave if they already matched to `1` to let it happen anyway :thinking:

**Comment by agusmoles** [comment]
I think it's working great on my end! I am not a good tester with only 2 people in my slack btw hahaha. Awesome improvement, much simpler :rocket:

File: test/matching.js (hunk 2 above)
Comment (kelostrada) [reply test/matching.js:6]: thx, I always forget about this feature ;)
Although I'm not a fan of arrow functions when using them as definitions

**Pull request merged**